config/opal_check_pc.m4

```diff
@@ -0,0 +1,105 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2021 IBM Corporation.  All rights reserved.
+dnl
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+
+dnl OPAL_CHECK_PKG_CONFIG
+dnl Check for availability of pkg-config and store the result.
+dnl If it is not available, store any passed in libs from the
+dnl --with-extra-libs configure option, or the known defaults.
+dnl
+dnl If it is available, allow configury to check for .pc files
+dnl and append to OMPI_WRAPPER_EXTRA_LIBS.
+AC_DEFUN([OPAL_CHECK_PKG_CONFIG], [
+  AS_IF([test -z "$OPAL_HAS_PKG_CONFIG"],
+        [
+          AC_MSG_CHECKING([If pkg-config is available])
+          which pkg-config > /dev/null 2>&1
+          AS_IF([test "$?" -eq "0"],
+                [
+                  AC_MSG_RESULT([yes])
+                  OPAL_HAS_PKG_CONFIG="yes"
+                ],
+                [
+                  AC_MSG_RESULT([no])
+                  OPAL_HAS_PKG_CONFIG="no"
+                  OPAL_FLAGS_APPEND_UNIQ([OMPI_WRAPPER_EXTRA_LIBS], [$OPAL_EXTRA_LIBS])
+                ]
+          )
+        ]
+  )
+])
+
+dnl OPAL_CHECK_LFLAGS
+dnl Get the -l flags using pkg-config on the passed in .pc file, and
+dnl append them to OMPI_WRAPPER_EXTRA_LIBS.
```

Comment:
@jsquyres

Reply:
I think opal_wrapper_extra_libs is probably marginally more correct, since many of the symbols are used in OPAL.  Point (2) isn't really valid.  In either case, the libraries aren't linked into the libmpi.a file, they're dependencies the user must specify.  And since libmpi.a depends on libopen-pal.so, any transitive inclusion into ABI you can come up with for static libraries means the impact on ABI is the same.